dragonchain/lib/dto/bnb.py

```diff
@@ -0,0 +1,372 @@
+        current_block = response["result"]["block"]["header"]["height"]
+        return int(current_block)
+
+    def should_retry_broadcast(self, last_sent_block: int) -> bool:
+        """Check whether a new broadcast should be attempted, given a number of blocks past (for L5)
+        Args:
+            last_sent_block: The block when the transaction was last attempted to be sent
+        Returns:
+            Boolean whether a broadcast should be re-attempted
+        """
+        return self.get_current_block() - int(last_sent_block) > BLOCK_THRESHOLD
+
+    def get_network_string(self) -> str:
+        """Get the network string for this blockchain. This is what's included in l5 blocks or sent to matchmaking
+        Returns:
+            Network string
+        """
+        return f"binance {'testnet' if self.testnet else 'mainnet'}"
+
+    def get_private_key(self) -> str:
+        """Get the private key for this network
+        Returns:
+            string of the private key
+        """
+        return self.b64_private_key
+
+    # https://docs.binance.org/api-reference/api-server.html#apiv1accountaddress
+    def _fetch_account(self):
+        """Fetch the account metadata for an address
+        Returns:
+            response containing account metadata
+        """
+        _log.info(f"[BINANCE] Fetching address metadata for {self.address}")
+        path = f"account/{self.address}"  # params expected inside path string
+        try:
+            response = self._call_node_api(path)
+            return response
+        except exceptions.InterchainConnectionError:
+            _log.warning("[BINANCE] Non 200 response from Binance node. May have been a 500 Bad Request or a 404 Not Found.")
+            return 0
+
+    def _build_transaction_msg(self, account_response: Dict[str, Any], transaction_payload: str) -> Dict:
+        # easier to just use from-addy for the to-addy than create a properly formatted dummy addy
+        inputs = {"address": self.address, "coins": [{"amount": 1, "denom": "BNB"}]}  # cannot send 0 amount
+        outputs = {"address": self.address, "coins": [{"amount": 1, "denom": "BNB"}]}
+        response = self._fetch_account()
+
+        transaction_data = {
+            "account_number": response["account_number"],
+            "sequence": response["sequence"],
+            "from": self.address,
+            "memo": transaction_payload,
+            "msgs": [{"type": "cosmos-sdk/Send", "inputs": [inputs], "outputs": [outputs]}],
+        }
+        return transaction_data
+
+    def sign_transaction(self, raw_transaction: Dict[str, Any]) -> str:
+        """Sign a transaction for this network
+        Args:
+            raw_transaction: The dictionary of the raw transaction containing:
+                to: hex string of the to address
+                symbol: the exchange symbol for the token (NOT defaulted to BNB)
+                amount:  amount of token in transaction
```

Comment:
please annotate they types of these values. Like I don't know if `amount` should be a hex string, regular string of numbers, integer, float, etc

Reply:
Fixed.